tests/components/UserActions/UserActions.spec.ts

```diff
@@ -417,4 +417,80 @@ describe('UserActions', () => {
             });
         });
     });
+
+    describe('ViewedByCustomer', () => {
+        let result;
+        let resultElementRow;
+        let resultElementCol;
+
+        beforeEach(() => {
+            result = Fake.createFakeResult();
+
+            resultElementRow = document.createElement('div');
+            resultElementRow.classList.add('coveo-result-row');
+
+            resultElementCol = document.createElement('div');
+            resultElementCol.classList.add('coveo-result-cell');
+
+            resultElementRow.appendChild(resultElementCol);
+        });
+
+        it('If the viewedByCustomer option is true, it should add a CiewedByCustomer Component', () => {
+            const mock = Mock.advancedComponentSetup<UserActions>(
+                UserActions,
+                new Mock.AdvancedComponentSetupOptions(null, { userId: 'testUserId' }, env => {
+                    fakeUserProfileModel(env.root, sandbox).getActions.returns(new Promise(() => {}));
+                    return env;
+                })
+            );
+
+            const resultArgs = { result: result, item: resultElementRow };
+            Coveo.$$(mock.env.root).trigger(ResultListEvents.newResultDisplayed, resultArgs);
+
+            return delay(() => {
+                expect(mock.cmp.options.viewedByCustomer).toBe(true);
+                expect(resultElementRow.getElementsByClassName('CoveoViewedByCustomer').length).toBe(1);
+            });
+        });
+
+        it('If the viewedByCustomer option is false, it should not add a ViewedByCustomer Component', () => {
+            const mock = Mock.advancedComponentSetup<UserActions>(
+                UserActions,
+                new Mock.AdvancedComponentSetupOptions(null, { userId: 'testUserId', viewedByCustomer: false }, env => {
+                    fakeUserProfileModel(env.root, sandbox).getActions.returns(new Promise(() => {}));
+                    return env;
+                })
+            );
+
+            const resultArgs = { result: result, item: resultElementRow };
+            Coveo.$$(mock.env.root).trigger(ResultListEvents.newResultDisplayed, resultArgs);
+
+            return delay(() => {
+                expect(mock.cmp.options.viewedByCustomer).toBe(false);
+                expect(resultElementRow.getElementsByClassName('CoveoViewedByCustomer').length).toBe(0);
+            });
+        });
+
+        it('If the div already has ViewedByCustomer component it should not add another', () => {
```

Comment:
I'd put this in the describe 'if viewedByCustomer option is set to true', and change the text of it to `it should not add a ViewedByCustomer component if ones is already there`